### PR TITLE
fix(AsuraScans): remake the presence to match asura website

### PR DESCRIPTION
**Pull request by rois2coeurs**
## Description 

- Added the new URL.
- Fixed regex for URL to accommodate Asura Scans' new paths.
- Added fetching for comic covers.
- Fixed the progress calculation issue.
- Fixed the issue with retrieving the comic title.
- Fixed the issue with retrieving the current chapter.
- Added settings to the presence:
  - Privacy Mode: Only displays that the user is on Asura Scans.
  - Display Current Chapter.
  - Display Reading Percentage.
  - Show Cover.
  - Show Buttons.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image

</details>

### CODE REVIEW

File: websites/A/Asura Scans/presence.ts [hunk 1]
```diff
@@ -1,54 +1,88 @@
...
 			largeImageKey:
 				"https://cdn.rcd.gg/PreMiD/websites/A/Asura%20Scans/assets/logo.png",
-		};
+			type: ActivityType.Watching,
+		},
+		displayPercentage = await presence.getSetting<boolean>("readingPercentage"),
+		privacyMode = await presence.getSetting<boolean>("privacy"),
+		displayChapter = await presence.getSetting<boolean>("chapterNumber"),
+		displayCover = await presence.getSetting<boolean>("showCover"),
+		displayButtons = await presence.getSetting<boolean>("showButtons");
```
Comment (theusaf) [inline websites/A/Asura Scans/presence.ts:28]: For better efficiency, combine this using `Promise.all` and destructuring.

File: websites/A/Asura Scans/metadata.json [hunk 2]
```diff
@@ -31,5 +39,51 @@
 		"manga",
 		"manhua",
 		"manhwa"
+	],
+	"readLogs": true,
```
Comment (theusaf) [inline websites/A/Asura Scans/metadata.json:43]: This presence does not read logs.
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/A/Asura Scans/presence.ts (hunk 1 above)
Comment (rois2coeurs) [reply websites/A/Asura Scans/presence.ts]: Will just the Promise.all do the destructing ?

**Review by theusaf — APPROVED** [review]

File: websites/A/Asura Scans/presence.ts [hunk 3]
```diff
@@ -1,54 +1,96 @@
...
+}
+
+const comic = new Comic();
+
+presence.on("UpdateData", async () => {
 	const { pathname, href } = window.location,
 		presenceData: PresenceData = {
 			startTimestamp: browsingTimestamp,
 			largeImageKey:
 				"https://cdn.rcd.gg/PreMiD/websites/A/Asura%20Scans/assets/logo.png",
```
Comment (Bas950) [inline websites/A/Asura Scans/presence.ts:21]:
```suggestion
			largeImageKey: ASURA_SCANS_LOGO,
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
For the rest LGTM

**Review by Bas950 — APPROVED** [review]

**Pull request merged**